Detector/DetComponents/src/CompareTrackHitPositionAndCellId.cpp

```diff
@@ -0,0 +1,132 @@
+}
+
+StatusCode CompareTrackHitPositionAndCellId::initialize() { return GaudiAlgorithm::initialize(); }
+
+StatusCode CompareTrackHitPositionAndCellId::execute() {
+  auto lcdd = m_geoSvc->lcdd();
+  auto allReadouts = lcdd->readouts();
+  auto readoutBarrel = lcdd->readout("TrackerBarrelReadout");
+  auto m_decoderBarrel = readoutBarrel.idSpec().decoder();
+  auto readoutEndcap = lcdd->readout("TrackerEndcapReadout");
+  auto m_decoderEndcap = readoutEndcap.idSpec().decoder();
+  auto segmentationXZ =
+      dynamic_cast<DD4hep::DDSegmentation::CartesianGridXZ*>(readoutBarrel.segmentation().segmentation());
+  if (nullptr == segmentationXZ) {
+    error() << "Could not retrieve segmentation!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  double l_segGridSizeXBarrel = segmentationXZ->gridSizeX();
+  double l_segGridSizeZBarrel = segmentationXZ->gridSizeZ();
+
+  segmentationXZ = dynamic_cast<DD4hep::DDSegmentation::CartesianGridXZ*>(readoutEndcap.segmentation().segmentation());
+  if (nullptr == segmentationXZ) {
+    error() << "Could not retrieve segmentation!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  double l_segGridSizeXEndcap = segmentationXZ->gridSizeX();
+  double l_segGridSizeZEndcap = segmentationXZ->gridSizeZ();
+  std::cout << l_segGridSizeXBarrel << std::endl;
+
+  DD4hep::Geometry::VolumeManager volman = m_geoSvc->lcdd()->volumeManager();
+
+  const fcc::PositionedTrackHitCollection* hits = m_positionedTrackHits.get();
+  double fcc_l1 = 0;
+  double fcc_l2 = 0;
+  double l_tolerance = 10;
+  for (auto hit : (*hits)) {
+    long long int theCellId = hit.core().cellId;
+    debug() << theCellId << endmsg;
+    m_decoderBarrel->setValue(theCellId);
+    int system_id = (*m_decoderBarrel)["system"];
+    if (system_id == 0 || system_id == 1) {
+
+      auto detelement = volman.lookupDetElement(theCellId);
+      const auto& transformMatrix = detelement.worldTransformation();
+      double outGlobal[3];
+      fcc_l1 = (*m_decoderBarrel)["x"] * l_segGridSizeXBarrel;
+      fcc_l2 = (*m_decoderBarrel)["z"] * l_segGridSizeZBarrel;
+      double inLocal[] = {fcc_l1, 0, fcc_l2};
+      transformMatrix.LocalToMaster(inLocal, outGlobal);
+      auto edmPos = fcc::Point();
+      edmPos.x = outGlobal[0] / dd4hep::mm;
+      edmPos.y = outGlobal[1] / dd4hep::mm;
+      edmPos.z = outGlobal[2] / dd4hep::mm;
+
+      debug() << " hit in system: " << system_id;
+      int layer_id = (*m_decoderBarrel)["layer"];
+      debug() << "\t layer " << layer_id;
+      int module_id = (*m_decoderBarrel)["module"];
+      debug() << "\t module " << module_id;
+      debug() << endmsg;
+
+      debug() << "Hit with Cell Id: " << theCellId << endmsg;
+      debug() << "position: x: " << hit.position().x << "\t y: " << hit.position().y << "\t z: " << hit.position().z
+              << endmsg;
+      debug() << "position: x: " << edmPos.x << "\t y: " << edmPos.y << "\t z: " << edmPos.z << endmsg;
```

Comment:
Could you please distinguish in the debug output which position is the hit positions and which is the edm position.